coordination/options/options.go

```diff
@@ -0,0 +1,173 @@
+		c.SessionStartTimeout = timeout
+	}
+}
+
+// WithSessionKeepAliveTimeout returns an OpenSessionOption that specifies the time that the client will wait before it
+// terminates the gRPC stream and tries to reconnect if no successful responses have been received from the server.
+//
+// If this is not set, the client uses the default time 10 seconds.
+func WithSessionKeepAliveTimeout(timeout time.Duration) OpenSessionOption {
+	return func(c *OpenSessionOptions) {
+		c.SessionKeepAliveTimeout = timeout
+	}
+}
+
+// WithSessionReconnectDelay returns an OpenSessionOption that specifies the time that the client will wait before it
+// tries to reconnect the underlying gRPC stream in case of error.
+//
+// If this is not set, the client uses the default time 500 milliseconds.
+func WithSessionReconnectDelay(delay time.Duration) OpenSessionOption {
+	return func(c *OpenSessionOptions) {
+		c.SessionReconnectDelay = delay
+	}
+}
+
+// OpenSessionOption configures how we open a new session.
+type OpenSessionOption func(c *OpenSessionOptions)
+
+// OpenSessionOptions configure an OpenSession call. OpenSessionOptions are set by the OpenSessionOption values passed
+// to the OpenSession function.
+type OpenSessionOptions struct {
+	Description             string
+	SessionTimeout          time.Duration
+	SessionStartTimeout     time.Duration
+	SessionStopTimeout      time.Duration
+	SessionKeepAliveTimeout time.Duration
+	SessionReconnectDelay   time.Duration
+}
+
+// WithEphemeral returns an AcquireSemaphoreOption that causes to create an ephemeral semaphore.
+//
+// Ephemeral semaphores are created with the first acquire operation and automatically deleted with the last release
+// operation. Ephemeral semaphore are always created with the limit of coordination.MaxSemaphoreLimit.
+func WithEphemeral(ephemeral bool) AcquireSemaphoreOption {
+	return func(c *Ydb_Coordination.SessionRequest_AcquireSemaphore) {
+		c.Ephemeral = ephemeral
+	}
+}
+
+// WithAcquireTimeout returns an AcquireSemaphoreOption which sets the timeout after which the operation fails if it
+// is still waiting in the queue. Use 0 to make the AcquireSemaphore method fail immediately if the semaphore is already
+// acquired by another session.
+//
+// If this is not set, the client waits for the acquire operation result until the operation or session context is done.
+// You can reset the default value of this timeout by calling the WithNoAcquireTimeout method.
+func WithAcquireTimeout(timeout time.Duration) AcquireSemaphoreOption {
+	return func(c *Ydb_Coordination.SessionRequest_AcquireSemaphore) {
+		c.TimeoutMillis = uint64(timeout.Milliseconds())
+	}
+}
+
+// WithNoAcquireTimeout returns an AcquireSemaphoreOption which disables the timeout after which the operation fails if
+// it is still waiting in the queue.
+//
+// This is the default behavior. You can set the specific timeout by calling the WithAcquireTimeout method.
+func WithNoAcquireTimeout() AcquireSemaphoreOption {
```

Comment:
NoTimeout can mean two things:
1. Wait infinite time
2. No wait and return immediately

What about rename the method to WithAquireInfiniteTimeout?

Reply:
I think it not useful anymore since there is the MaxTimeout constant in the `coordination` package.